docs/concepts.md

```diff
@@ -0,0 +1,12 @@
+### Federation V2 Concepts
+
+| Concept | Description |
+|-----------|-------------|
+| Federate | Federating a set of Kubernetes clusters means, effectively creating a common interface to the pool of these federated clusters which can be used to deploy kubernetes applications across the same. |
+| Federation | Kubernetes Cluster Federation enables users to federate multiple Kubernetes clusters for resources distribution, service discovery, high availability etc across multiple clusters. |
+| Host Cluster | A cluster which is used to expose the federation API and run the federation control plane. |
+| Member Cluster | A cluster which is registered with the federation API and that federation controllers have authentication credentials for. The Host Cluster can also be a Member Cluster. |
+| ServiceServiceRecord | A resource that associates one or more Kubernetes Service resources and how to access the Service, with a scheme for constructing Domain Name System (DNS) [resource records](https://www.ietf.org/rfc/rfc1035.txt) for the Service. |
+| IngressDNSRecord | A resource that associates one or more Kubernetes Ingress and how to access the Ingress resources, with a scheme for constructing Domain Name System (DNS) [resource records](https://www.ietf.org/rfc/rfc1035.txt) for the Ingress. |
+| DNSEndpoint | A [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) wrapper for the Endpoint resource. |
```

Comment:
You may want to refer to the other resources in this list as "Custom Resource".

Reply:
Can you elaborate more for this question?

Reply:
Personally, I'm not sure it will add a lot to identify custom resources, since everything in our APIs is a custom resource :)

Reply:
@pmorie do you mean we can actually delete those concepts from the doc? like ServiceDNSRecord, IngressDNSRecord, DNSEndpoint, Endpoint etc?